### PR TITLE
mds,client: correct last_issue struct fields and consistently send last_issue in cap messages

### PR DESCRIPTION
**Pull request by batrick**
First set of changes I'd like to do for

https://tracker.ceph.com/issues/66704

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/68629](https://tracker.ceph.com/issues/68629).

**Comment by batrick** [comment]
https://tracker.ceph.com/issues/68629#note-1

**Comment by vshankar** [comment]
(On this today)

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/68786](https://tracker.ceph.com/issues/68786).

**Review by vshankar — APPROVED** [review]

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/68859](https://tracker.ceph.com/issues/68859).

**Comment by batrick** [comment]
https://tracker.ceph.com/issues/68859#note-2

File: src/client/Client.cc [hunk 1]
```diff
@@ -5518,10 +5518,10 @@ void Client::handle_cap_export(MetaSession *session, Inode *in, const MConstRef<
         if (it != in->caps.end()) {
 	  Cap &tcap = it->second;
 	  if (tcap.cap_id == m->peer.cap_id &&
-	      ceph_seq_cmp(tcap.seq, m->peer.seq) < 0) {
+	      ceph_seq_cmp(tcap.seq, m->peer.issue_seq) < 0) {
 	    tcap.cap_id = m->peer.cap_id;
-	    tcap.seq = m->peer.seq - 1;
-	    tcap.issue_seq = tcap.seq;
+	    tcap.seq = m->peer.issue_seq - 1;
+	    tcap.issue_seq = tcap.issue_seq;
```
Comment (idryomov) [inline src/client/Client.cc:5524]: @batrick I was taking a look at "ceph: correct ceph_mds_cap_peer field name" patch for the kernel client and there is something fishy here:
- previously both clients effectively assigned `m->peer.seq - 1` to `tcap.seq` and `tcap.issue_seq`
- `m->peer.seq` got renamed to `m->peer.issue_seq`
- kernel client continues to assign that value to both fields
```
			tcap->seq = t_issue_seq - 1;
			tcap->issue_seq = t_issue_seq - 1;
```
- userspace client assigns that value only to `tcap.seq`, for `tcap.issue_seq` there appears to be a bogus self-assignment

Comment (batrick) [reply src/client/Client.cc:5524]: > @batrick I was taking a look at "ceph: correct ceph_mds_cap_peer field name" patch for the kernel client and there is something fishy here:
> 
>     * previously both clients effectively assigned `m->peer.seq - 1` to `tcap.seq` and `tcap.issue_seq`

Correct.

>     * `m->peer.seq` got renamed to `m->peer.issue_seq`

Correct.

>     * kernel client continues to assign that value to both fields
> 
> 
> ```
> 			tcap->seq = t_issue_seq - 1;
> 			tcap->issue_seq = t_issue_seq - 1;
> ```
> 
>     * userspace client assigns that value only to `tcap.seq`, for `tcap.issue_seq` there appears to be a bogus self-assignment

Oh, very nice catch. My find/replace clearly messed that up.

**Pull request merged**